lib/puppet/provider/service/launchd.rb

```diff
@@ -66,12 +66,25 @@ def self.launchd_paths
     ]
   end
 
+  # Gets the current darwin version, example 10.6 returns 9 and 10.10 returns 14
```

Comment:
Linking to https://en.wikipedia.org/wiki/Darwin_(operating_system) might be helpful here.